packages/lib-core/src/types/k8s.ts

```diff
@@ -0,0 +1,122 @@
+export type K8sVerb =
+  | 'create'
+  | 'get'
+  | 'list'
+  | 'update'
+  | 'patch'
+  | 'delete'
+  | 'deletecollection'
+  | 'watch';
+
+export enum Operator {
+  Exists = 'Exists',
+  DoesNotExist = 'DoesNotExist',
+  In = 'In',
+  NotIn = 'NotIn',
+  Equals = 'Equals',
+  NotEqual = 'NotEqual',
+  GreaterThan = 'GreaterThan',
+  LessThan = 'LessThan',
+  NotEquals = 'NotEquals',
+}
+
+export type MatchExpression = {
+  key: string;
+  operator: Operator | string;
+  values?: string[];
+  value?: string;
+};
+
+export type MatchLabels = {
+  [key: string]: string;
+};
+
+export type Selector = {
+  matchLabels?: MatchLabels;
+  matchExpressions?: MatchExpression[];
+};
+
+enum BadgeType {
+  DEV = 'Dev Preview',
+  TECH = 'Tech Preview',
+}
+
+export type K8sModel = {
+  abbr: string;
+  kind: string;
+  label: string;
+  labelKey?: string;
+  labelPlural: string;
+  labelPluralKey?: string;
+  plural: string;
+  propagationPolicy?: 'Foreground' | 'Background';
+
+  id?: string;
+  crd?: boolean;
+  apiVersion: string;
+  apiGroup?: string;
+  namespaced?: boolean;
+  selector?: Selector;
+  labels?: { [key: string]: string };
+  annotations?: { [key: string]: string };
+  verbs?: K8sVerb[];
+  shortNames?: string[];
+  badge?: BadgeType;
+  color?: string;
+
+  // Legacy option for supporing plural names in URL paths when `crd: true`.
+  // This should not be set for new models, but is needed to avoid breaking
+  // existing links as we transition to using the API group in URL paths.
+  legacyPluralURL?: boolean;
+};
+
+export type DiscoveryResources = {
+  adminResources: string[];
+  allResources: string[];
+  configResources: K8sModel[];
+  clusterOperatorConfigResources: K8sModel[];
+  models: K8sModel[];
+  namespacedSet: Set<string>;
+  safeResources: string[];
+  groupVersionMap: {
+    [key: string]: {
+      versions: string[];
+      preferredVersion: string;
+    };
+  };
+};
+
+export type OwnerReference = {
+  name: string;
+  kind: string;
+  uid: string;
+  apiVersion: string;
+  controller?: boolean;
+  blockOwnerDeletion?: boolean;
+};
+
+export type ObjectMetadata = {
+  annotations?: { [key: string]: string };
+  clusterName?: string;
+  creationTimestamp?: string;
+  deletionGracePeriodSeconds?: number;
+  deletionTimestamp?: string;
+  finalizers?: string[];
+  generateName?: string;
+  generation?: number;
+  labels?: { [key: string]: string };
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  managedFields?: any[];
+  name?: string;
+  namespace?: string;
+  ownerReferences?: OwnerReference[];
+  resourceVersion?: string;
+  uid?: string;
+};
+
+// Properties common to (almost) all Kubernetes resources.
+export type K8sResourceCommon = {
+  apiVersion?: string;
+  kind?: string;
+  metadata?: ObjectMetadata;
+};
```

Comment:
This is being added by @vojtechszocs's PR - https://github.com/openshift/dynamic-plugin-sdk/pull/30/files#diff-01541dad1d9bd6352f582e703d8f5a8e5c2b81b6b92e7cab86fd07d542c5d8b6R12